lib/renaissance/auctions/auctions.ex

```diff
@@ -1,50 +1,51 @@
 defmodule Renaissance.Auctions do
   import Ecto.Query
-  alias Renaissance.{Repo, Auction}
+  alias Renaissance.{Auction, Repo, Bids}
+  alias Renaissance.Helpers.{Adapt, Compare}
 
   def create_auction(user_id, params) do
-    details =
-      if Map.has_key?(params, "auction") do
-        params["auction"]
-      else
-        params
-      end
-
-    details =
-      details
-      |> format_price()
+    params =
+      params
+      |> Adapt.format_amount("price")
       |> Map.put("seller_id", user_id)
 
-    Repo.insert(Auction.changeset(%Auction{}, details))
+    Auction.changeset(%Auction{}, params)
+    |> Repo.insert()
   end
 
-  def get(id) do
+  def exists?(id) do
+    Repo.exists?(from a in Auction, where: a.id == ^id)
+  end
+
+  def get!(id) do
     Auction
     |> preload(:seller)
     |> Repo.get!(id)
   end
 
+  def get_seller_id(id) do
+    if exists?(id), do: get!(id).seller_id
```

Comment:
There seems to be this pattern of `exists?(thing), do: the_operation(thing)`, which isn't really necessary, since you can just handle nil from get:

```
case get(id) do
  nil -> nil
  auction -> auction.seller_id
end
```

Reply:
IMO, should this even be checking it? Shouldn't checking for record existing be higher up the handling order?

Reply:
Where do you mean by higher up? The form?

If yes, the form errors (beyond those that can be checked using HTML tags) are generated by the changeset validation. I'm unaware of how else to generate them. All the examples I've seen generate form errors by means of the changeset validation; that said, I have not researched this topic in any depth.